rotkehlchen/chain/ethereum/transactions.py

```diff
@@ -52,41 +73,57 @@ def query_for_graph_delegation_txns(self, addresses: Sequence[ChecksumEvmAddress
                 event_subtypes=[HistoryEventSubType.APPROVE],
+                location_labels=[address],
             )
             dbevents = DBHistoryEvents(self.database)
             with dbevents.db.conn.read_ctx() as cursor:
-                if len(dbevents.get_history_events(
-                        cursor=cursor,
-                        filter_query=db_filter,
-                        has_premium=True,
-                )) == 0:
+                events = dbevents.get_history_events(
+                    cursor=cursor,
+                    filter_query=db_filter,
+                    has_premium=True,
+                )
+                if len(events) == 0:
                     continue
 
-            with self.database.conn.read_ctx() as cursor:
-                from_block = staking_contract.deployed_block
+                # find the earliest delegation to avoid querying a wide range
+                earliest_event = min(events, key=lambda event: event.timestamp)
+                from_block = DBEvmTx(self.database).get_transaction_block_by_hash(
+                    cursor=cursor,
+                    tx_hash=earliest_event.tx_hash,
+                ) or staking_contract.deployed_block
+
                 if (result := self.database.get_dynamic_cache(
                     cursor=cursor,
                     name=DBCacheDynamic.LAST_BLOCK_ID,
                     location=self.evm_inquirer.chain_name,
-                    location_name=LAST_GRAPH_DELGATIONS,
+                    location_name=LAST_GRAPH_DELEGATIONS,
                     account_id=address,
                 )) is not None:
                     from_block = result
 
-            to_block = self.evm_inquirer.get_latest_block_number()
-            events = self.evm_inquirer.get_logs(
+            log.debug(
+                f'Found thegraph delegation events for address {address} at '
+                f'{earliest_event.tx_hash.hex()}. Starting query from block {from_block}',
+            )
+            target_block = self.evm_inquirer.get_latest_block_number()
+            log_events = self.evm_inquirer.get_logs(
                 contract_address=staking_contract.address,
                 abi=GRAPH_DELEGATION_TRANSFER_ABI,
                 event_name='DelegationTransferredToL2',
                 argument_filters={'l2Delegator': address},
                 from_block=from_block,
-                to_block=to_block,
+                to_block=target_block,
+                log_iteration_cb=self._graph_delegation_callback,
             )
 
-            if len(events) == 0:
-                log.info(f'No graph delegation events found for {address} from block {from_block} to {to_block}')  # noqa: E501
+            if len(log_events) == 0:
+                log.info(f'No graph delegation events found for {address} from block {from_block} to {target_block}')  # noqa: E501
             else:
-                for event in events:
+                log.info(
```

Comment:
So now you are mixing info and debug in this function. Either turn them all to debug or all info. I would say all debug.